### PR TITLE
New service netbox

### PR DESCRIPTION
**Pull request by balcirakpeter**
- gen script fetches resources and their assigned members. For resources
  fetches netboxGroupName which serves as a group name in the destination
  and netboxIsStaff which indicates is_staff value for its members.
  Resources without the name attribute are skipped. We fetch name, email
  login, for each user. is_active is set to true and is_staff to false. These two
  are attributes in netbox. We set them in gen script in case, we will
  fill them from perun in the future. groups list is filled with
  netboxGroupName attribute for each member's resource.
- send script creates missing groups first. Then creates missing users.
  In this part we generate also password and set group membership. In
  the end we update users. If user does not exists in perun but exists
  in netbox he is set to inactive and his group list is erased. If he
  exists both in perun and netbox, his attributes are updated except
  password. laso group membership is updated in this part. We do not
  delete users and groups. When group does not exists in perun but
  exists in netbox. It will be empty in the end of the process.
- for this service is needed file with name netbox in /etc/perun/, which
  contains authorization token. Destination needs to be set to without
  trailing "/".

### CODE REVIEW

File: send/netbox [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+# callServer
+#-----------------------
+#Parameters:
+# url     - url of server and method address to call
+# type    - GET, POST or PATCH
+# content - hash content of request, will be encoded to json output
+#-----------------------
+#Returns:
+# resolved status of call
+# 0 means everything is ok and done
```
Comment (cuadradek) [inline send/netbox:155]: I would change this to something like: Returns hash of response content

File: send/netbox [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	return check_server_response( $server_response );
+}
+
+#Name:
+# createConnection
+#-----------------------
+#Parameters:
+# url     - url of server and method address to call
+# type    - GET, POST or PATCH
+# content - hash content of request, will be encoded to json output
```
Comment (cuadradek) [inline send/netbox:180]: content is already passed as json

File: send/netbox [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+#-----------------------
+sub create_connection {
+	my $url = shift;
+	my $type = shift;
+	my $content = shift;
+
+	my $headers = HTTP::Headers->new;
+	$headers->header('Content-Type' => 'application/json');
+	$headers->header('Accept' => 'application/json,text/json');
+	$headers->header('Authorization' => 'Token ' . $TOKEN);                                                       #ASI DOBRE
```
Comment (cuadradek) [inline send/netbox:196]: You probably forgot to delete the comment

**Review by cuadradek — CHANGES_REQUESTED** [review]
Looks good to me, just a couple of unclarities in comments

File: send/netbox (hunk 2 above)
Comment (balcirakpeter) [reply send/netbox]: Fixed

File: send/netbox (hunk 3 above)
Comment (balcirakpeter) [reply send/netbox]: Fiked

File: send/netbox (hunk 1 above)
Comment (balcirakpeter) [reply send/netbox]: Fixed

**Review by cuadradek — APPROVED** [review]

**Comment by rosinec** [comment]
Provisioning scripts are well written and meets all requirements that were desired by service provider. 
Provisioning process was tested with the service provider.

**Review by rosinec — APPROVED** [review]
Provisioning scripts are well written and meets all requirements that were desired by service provider.
Provisioning process was tested with the service provider.

**Pull request merged**